scripts/masternode.js

```diff
@@ -0,0 +1,217 @@
+"use strict";
+
+class Masternode {
+    static protocolVersion = 70926;
+    constructor({walletPrivateKey, mnPrivateKey, collateralTxId, outidx, addr} = {}) {
+	this.walletPrivateKey = walletPrivateKey;
+	this.mnPrivateKey = mnPrivateKey;
+	this.collateralTxId = collateralTxId;
+	this.outidx = outidx;
+	this.addr = addr;
+    }
+
+    async getStatus() {
+	const url= `${cNode.url}/listmasternodes?params=${this.collateralTxId}`;
+	try{
+	    const masternodes = (await (await fetch(url)).json()).filter(m=>m.outidx === this.outidx);
+	    if(masternodes.length > 0) {
+		return masternodes[0].status;
```

Comment:
Since this function is frequently used for display purposes, we need to sanitise what it returns.

Reply:
Good catch, totally forgot that